packages/manager/modules/pci/src/projects/project/data-processing/submit-job/submit-job.controller.js

```diff
@@ -0,0 +1,147 @@
+    );
+    const version = find(
+      engine.availableVersions,
+      (v) => v.name === this.state.jobEngine.version,
+    );
+    this.regions = version.availableRegions.map((region) => ({
+      name: region,
+      hasEnoughQuota: () => true,
+    }));
+  }
+
+  /**
+   * Handler for region selector change
+   * @param name Name of the selected region
+   */
+  onChangeRegionHandler({ name }) {
+    this.state.region = name;
+  }
+
+  /**
+   * Handler for job type selector job
+   * @param jobType Selected job type
+   */
+  onChangeJobTypeHandler(jobType) {
+    const e = find(this.capabilities, (o) => o.name === jobType.engine);
+    this.state.jobEngine = {
+      ...jobType,
+      templates: e.templates,
+    };
+  }
+
+  onSubmitJobSizingHandler() {
+    // trigger job sizing component values update
+    this.jobSizingValidate = !this.jobSizingValidate;
+  }
+
+  /**
+   * Handler for job config change
+   * @param jobConfig Job configuration
+   */
+  onChangeJobConfigHandler(jobConfig) {
+    this.state.jobConfig = jobConfig;
+  }
+
+  onSubmitJobHandler() {
+    this.isSubmitting = true;
+    let args = '';
+    if (this.state.jobConfig.arguments.length > 0) {
+      args = this.state.jobConfig.arguments.map((o) => o.title).join(',');
+    }
+    const payload = {
+      containerName: this.state.jobConfig.swiftContainer,
+      engine: this.state.jobEngine.engine,
+      engineVersion: this.state.jobEngine.version,
+      name: this.state.jobConfig.jobName,
+      region: this.state.region,
+      engineParameters: [
+        {
+          name: 'main_application_code',
+          value: this.state.jobConfig.mainApplicationCode,
+        },
+        {
+          name: 'arguments',
+          // handle iceberg limitation concerning arrays. We use comma-delimited string
+          value: args,
+        },
+        {
+          name: 'driver_memory',
+          value: convertMemory(`${this.state.jobSizing.driverMemoryGb}G`, 'Gi'),
+        },
+        {
+          name: 'executor_memory',
+          value: convertMemory(`${this.state.jobSizing.workerMemoryGb}G`, 'Gi'),
+        },
+        {
+          name: 'driver_cores',
+          value: this.state.jobSizing.driverCores.toString(),
+        },
+        {
+          name: 'executor_num',
+          value: this.state.jobSizing.workerCount.toString(),
+        },
+        {
+          name: 'executor_cores',
+          value: this.state.jobSizing.workerCores.toString(),
+        },
+        {
+          name: 'job_type',
+          value: this.state.jobConfig.jobType,
+        },
+      ],
+    };
+    if (this.state.jobConfig.jobType === 'java') {
+      payload.engineParameters.push({
+        name: 'main_class_name',
+        value: this.state.jobConfig.mainClass,
+      });
+    }
+    this.dataProcessingService.submitJob(this.projectId, payload).then(
+      () => {
+        this.goBack();
+      },
+      () => {
+        if (this.submitRetries < 2) {
```

Comment:
We don't do that, it would be weird to have this behaviour only on this part of the Manager 🙂 

When we will do this, it will be with a generic handling, so you don't have to do it everywhere